checkpoint.go

```diff
@@ -116,11 +119,11 @@ func Check(p *CheckParams) (*CheckResponse, error) {
 		p.OS = runtime.GOOS
 	}
 
-	// If we're given a SignatureFile, then attempt to read that.
+	// If we're not given a Signature, then attempt to read one.
 	signature := p.Signature
-	if p.Signature == "" && p.SignatureFile != "" {
+	if p.Signature == "" {
 		var err error
-		signature, err = checkSignature(p.SignatureFile)
+		signature, err = getSignature(p.SignatureFile)
```

Comment:
This is a bit odd, because now any `SignatureFile` that you specify is ignored (unless `getSystemUUID` happens to err). If you deleted `getSignature` and did this here instead:
```
    // If we're given a SignatureFile, then attempt to read that.
    signature := p.Signature
    if p.Signature == "" {
        var err error
        if p.SignatureFile == "" {
            signature, err = getSystemdUUID()
        } else {
            signature, err = checkSignature(p.SignatureFile)
        }
        if err != nil {
            return nil, err
        }
    }
```
Then a user can specify either a `Signature` or a `SignatureFile` to get the existing library behaviour (with `Signature` preferred if both are supplied as before) or omit both to use `getSystemUUID`.